File: tkpdtransaction/src/main/java/com/tokopedia/transaction/checkout/data/entity/request/ProductDataCheckoutRequest.java
package com.tokopedia.transaction.checkout.data.entity.request;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * @author anggaprasetiyo on 05/03/18.
 */

public class ProductDataCheckoutRequest {

    @SerializedName("product_id")
    @Expose
    public int productId;

    private ProductDataCheckoutRequest(Builder builder) {
        productId = builder.productId;
    }

    public static final class Builder {
        private int productId;

        public Builder() {
        }

        public Builder productId(int val) {
            productId = val;
            return this;
        }

        public ProductDataCheckoutRequest build() {
            return new ProductDataCheckoutRequest(this);
        }
    }
}

File: features/transaction/checkout/src/main/java/com/tokopedia/checkout/data/model/request/checkout/ProductDataCheckoutRequest.kt
package com.tokopedia.checkout.data.model.request.checkout

import android.annotation.SuppressLint
import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class ProductDataCheckoutRequest(
        @SuppressLint("Invalid Data Type")
        @SerializedName("product_id")
        var productId: Long = 0,
        @SerializedName("is_ppp")
        var isPurchaseProtection: Boolean = false,
        @SerializedName("product_quantity")
        var productQuantity: Int = 0,
        @SerializedName("product_notes")
        var productNotes: String? = null,

        var productName: String? = null,
        var productPrice: String? = null,
        var productBrand: String? = null,
        var productCategory: String? = null,
        var productVariant: String? = null,
        var productShopId: String? = null,
        var productShopType: String? = null,
        var productShopName: String? = null,
        var productCategoryId: String? = null,
        var productListName: String? = null,
        var productAttribution: String? = null,
        var cartId: Long = 0,
        var warehouseId: String? = null,
        var productWeight: String? = null,
        var promoCode: String? = null,
        var promoDetails: String? = null,
        var buyerAddressId: String? = null,
        var shippingDuration: String? = null,
        var courier: String? = null,
        var shippingPrice: String? = null,
        var codFlag: String? = null,
        var tokopediaCornerFlag: String? = null,
        var isFulfillment: String? = null,
        var isDiscountedPrice: Boolean = false,
        var isFreeShipping: Boolean = false,
        var isFreeShippingExtra: Boolean = false,
        var campaignId: Int = 0
) : Parcelable